src/deps_deploy/maven_settings.clj

```diff
@@ -0,0 +1,120 @@
+(ns deps-deploy.maven-settings
+  "Functions to work with maven settings.xml and settings-security.xml."
+  (:require [clojure.java.io :as io]
+            [clojure.string :as str])
+  (:import [org.apache.maven.settings.io.xpp3 SettingsXpp3Reader]
+           [org.sonatype.plexus.components.cipher
+            DefaultPlexusCipher]
+           [org.sonatype.plexus.components.sec.dispatcher
+            DefaultSecDispatcher
+            SecUtil]))
+
+(def default-settings-path (str (System/getProperty "user.home") "/.m2/settings.xml"))
+(def default-settings-security-path (str (System/getProperty "user.home") "/.m2/settings-security.xml"))
+
+;; Adapted from https://github.com/jelmerk/maven-settings-decoder/blob/master/src/main/java/org/github/jelmerk/maven/settings/Decoder.java
+
+(defn ^String decode-password
+  "Decodes a password (got from settings.xml)."
+  [^String encoded-password
+   ^String key]
+  (-> (DefaultPlexusCipher.)
+      (.decryptDecorated encoded-password key)))
+
+(defn ^String decode-master-password
```

Comment:
I think I'd prefer just inlining this fn, as the name it provides IMO doesn't provide too much value

Reply:
done.